File: api/schemas/competitor.py
from typing import Optional

from pydantic import BaseModel


class CompetitorBase(BaseModel):
    first_name: Optional[str]
    last_name: Optional[str]
    organization: Optional[str]
    location: Optional[str]


class CompetitorCreate(CompetitorBase):
    pass


class CompetitorUpdate(CompetitorBase):
    pass


class Competitor(CompetitorBase):
    id: int
    tournament_id: int

    class Config:
        orm_mode = True

File: frontend/src/models/competitor.js
export default class Competitor {
  constructor(id, firstName, lastName, organization, location) {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.organization = organization;
    this.location = location;
  }

  asCreateRequestBody() {
    return {
      first_name: this.firstName,
      last_name: this.lastName,
      organization: this.organization,
      location: this.location,
    };
  }

  static fromCreateResponseBody(response) {
    return new Competitor(
      response.id,
      response.first_name,
      response.last_name,
      response.organization,
      response.location,
    );
  }
}
